### ISSUE TITLE
Fix FP S3001 `no-delete-var`: ignore object optional properties

### ISSUE THREAD

**Issue by yassin-kammoun-sonarsource** [SonarSource/SonarJS#2994]
The rule S3001 raises false-positive issues when deleting object optional properties:

```javascript
let obj = { prop: possiblyUndefined() };
delete obj?.prop; // FP
```

Reported in https://community.sonarsource.com/t/typescript-s3001-false-positive-when-delete-operator-used-on-optional-properties-object/59200
